src/Codeception/Module/Db.php

```diff
@@ -542,6 +544,10 @@ private function connect($databaseKey, $databaseConfig)
             $options[\PDO::MYSQL_ATTR_SSL_VERIFY_SERVER_CERT] = (boolean) $databaseConfig[ 'ssl_verify_server_cert' ];
         }
 
+        if (array_key_exists('init_command', $databaseConfig) && defined('\PDO::MYSQL_ATTR_INIT_COMMAND')) {
+            $options[\PDO::MYSQL_ATTR_INIT_COMMAND] = $databaseConfig['init_command'];
+        }
```

Comment:
Is there any advantage of using mysql-specific init command over calling [executeQuery](https://github.com/Codeception/Codeception/blob/3.0.3/src/Codeception/Lib/Driver/Db.php#L285) in [connect ](https://github.com/Codeception/Codeception/blob/3.0.3/src/Codeception/Lib/Driver/Db.php#L36)method?

Reply:
No, I don't think so.
If we added executeQuery in the `__construct` of `Driver\Db` that would be much more flexible than depending on the mysql-specific `init_command`, and it would also give the possibility to solve https://github.com/Codeception/Codeception/issues/5605 as well by configuring queries for creating/using the database if it doesn't exist right after creating the PDO object. 
Adding executeQuery in the `connect` method like you offered wouldn't really work without extending the `Module\Db` class and overriding how `connectToDatabases/reconnectDatabases` works, unless I am missing something.

Reply:
I looked at this code again and I think that the best place to execute init_command is at the bottom of this method, either before or after this line:
```php
$this->debugSection('Db', 'Connected to ' . $databaseKey . ' ' . $this->drivers[$databaseKey]->getDb());`
```

Reply:
I agree with the location, but do you mean to execute `executeQuery` instead of `init_command`? Because `init_command` is the mysql specific command which is executed during the creation of PDO connection, and the PDO connection would have already been created a few lines before.

Reply:
mysql-specific connection attribute exists for legacy reasons - because mysql C lib and mysql command line client has it. It made sense to have it as a connection attribute, because mysql client and old PHP mysql extension could reconnect automatically if connection was lost. Is PDO even able to reconnect if connection is lost? I think that it throws exception instead, so there is no reason to use connection attribute.

Reply:
I'm okay with not using mysql-specific attributes for PDO connection.
Do you think a change like this would be accepted ? It would solve both of my issues.
Params array which is passed to `executeQuery` could be configured as well (shown as empty array in the following snippet)

```
if ($databaseConfig['waitlock']) {
    $this->_getDriver()->setWaitLock($databaseConfig['waitlock']);
}

// NEW CODE START
foreach ($databaseConfig['initial_queries'] as $initialQuery) {
    $this->drivers[$databaseKey]->executeQuery($initialQuery, []);
}
// NEW CODE END

$this->debugSection('Db', 'Connected to ' . $databaseKey . ' ' . $this->drivers[$databaseKey]->getDb());
```

Reply:
Yes, something like that could be accepted.

Please create a fresh branch from 3.0 to get builds passing
and implement a test for it in https://github.com/Codeception/Codeception/blob/3.0/tests/unit/Codeception/Module/Db/TestsForDb.php or one of database specific test files in that directory (I think that you will have to use `_reconfigure` to set your configuration options). Creating a separate test file is fine too, I think that it would look cleaner that way.

Reply:
I agree with @Naktibalda here :+1: